QuantLib/ql/experimental/credit/basecorrelationlossmodel.hpp

```diff
@@ -95,6 +95,9 @@ namespace QuantLib {
             const Handle<BaseCorrelationTermStructure<Corr2DInt_T> >& correlTS,
             const std::vector<Real>& recoveries,
             const typename BaseModel_T::copulaType::initTraits& traits = 
+#if defined(__GNUC__)
```

Comment:
Since `typename` seems to be the more correct syntax, I'd rather use a define that excludes the compilers that don't recognize it, i.e., something like

    #if !defined(QL_PATCH_MSVC90)

However, we might be able to sidestep the issue. Does something like the following work for you?

    private:
        typedef typename BaseModel_T::copulaType::initTraits initTraits;
    public:
        BaseCorrelationLossModel(
            const Handle<BaseCorrelationTermStructure<Corr2DInt_T> >& correlTS,
            const std::vector<Real>& recoveries,
            const initTraits& traits = initTraits()
            )

(It works for me on gcc 4.9, but I haven't checked on Visual C++.)